lib/Driver/ToolChains.cpp

```diff
@@ -1045,19 +1045,17 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
 
   addInputsOfType(Arguments, context.InputActions, types::TY_Object);
 
-  if (context.OI.DebugInfoKind > IRGenDebugInfoKind::LineTables) {
-    size_t argCount = Arguments.size();
-    if (context.OI.CompilerMode == OutputInfo::Mode::SingleCompile)
-      addInputsOfType(Arguments, context.Inputs, types::TY_SwiftModuleFile);
-    else
-      addPrimaryInputsOfType(Arguments, context.Inputs,
-                             types::TY_SwiftModuleFile);
-
-    if (Arguments.size() > argCount) {
-      assert(argCount + 1 == Arguments.size() &&
-             "multiple swiftmodules found for -g");
-      Arguments.insert(Arguments.end() - 1, "-add_ast_path");
-    }
+  size_t argCount = Arguments.size();
+  if (context.OI.CompilerMode == OutputInfo::Mode::SingleCompile)
+    addInputsOfType(Arguments, context.Inputs, types::TY_SwiftModuleFile);
+  else
+    addPrimaryInputsOfType(Arguments, context.Inputs,
+                           types::TY_SwiftModuleFile);
+  
+  addInputsOfType(Arguments, context.InputActions, types::TY_SwiftModuleFile);
+  
+  if (Arguments.size() > argCount) {
+    Arguments.insert(Arguments.end() - 1, "-add_ast_path");
```

Comment:
This has to be present before *every* swiftmodule input, so either there needs to be a more complicated transform here or the assertion needs to stay.